### DISCUSSION TITLE
CarlinKit 3.0? (red UI/AUTOKIT?)

### DISCUSSION THREAD

**Discussion by tokamac** [ludwig-v/wireless-carplay-dongle-reverse-engineering#96] [Q&A]
I am confused… on the CarlinKit website, **AUTOKIT** is said to be the former dongle compatible with both Apple CarPlay and Android Auto, that has the undetachable USB cable, and same form factor than **CARPLAY2AIR** (**CPLAY2AIR**). The newer version is called **CarlinKit 2.0** and is labelled "**CPC200-U2W-PLUS**", has the detachable USB-C to USB-A cable and is compatible with Apple CarPlay only.

But there seem to be an even newer version, unrelated to firmware updates, sometimes referred as "**CarlinKit 3.0**" on the CarlinKit official AliExpress store. But the Chinese, to simplify (sarcasm) keep the same reference for both on their plastic cases: CPC200-U2W-PLUS (maybe they were too lazy to make a change, or still have too many plastic boxes already etched with this reference in stock? I would not be surprised). In order to muddy the waters even more, Chinese style, they write in the description:

> "Carlinkit hereby declares that with regard to the "Carlinkit 3.0 Wireless CarPlay Activator" on the AliExpress: the product is the same product as Carlinkit 2.0. We just made a better quality chip to upgrade the original product."

So… it is the same but is is not the same. It is the same product, but we have changed the chip. How do these people think, Jesus!

Some people say that v2.0 (U2W) has a green UI for the configuration screen displayed on the car head unit, while v3.0 (still labelled U2W on the box but different actually) has a red UI and would be based on AUTOKIT. Does anyone know what is real and what is not? What is the benefit of AUTOKIT vs U2W if true. I thought AUTOKIT was a former solution in the first dongle, abandoned since?

**Comment by sunny2819** [comment]
CarlinKit 2.0 is often disconnected, very bad, the music is slightly stuck, the main controller is AT91SAM9260, it is very old, after sale is equal to no

**Comment by Tozzi89** [comment]
I would like to know this too, I'm not sure what firmware to use.

I have a CarlinKit 3.0 and the UI is red. It is named CPC200-U2W Plus with black carbon. What firmware should I use?

Maybe clarify the differences in README please.

**Reply by Tozzi89** [reply]
Update: I used file [U2W_AUTOKIT_Update_2022.06.17.1317.img](https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/blob/master/Firmware/U2W_AUTOKIT_Update_2022.06.17.1317.img), formatted a USB flash device to FAT32 and copied file over to it and renamed it to U2W_AUTOKIT_Update.img. Let it be connected to USB powerwall for ~5 minutes and it worked directly in the car afterwards! (Started car and let media player boot properly before attaching Carlinkit).

I had previous tried to upgrade the firmware through the webbrowser but it never came alive after update, only red LED.

CarlinKit 3.0, red UI, black carbon case and marked CPC200-U2W Plus.

The guide I used: https://carlinkitcarplay.com/blogs/wireless-dongle-adapter/u2w-u2w-plus-firmware-upgrade-by-usb-flash-drive

Hope this info helps!

**Comment by DrewHager571** [comment]
Anyone here tried the new Carlinkit 4.0? They advertise it as having both wireless AA and wireless Carplay. Any feedback from the ones who tried it if it is working ok?

https://autokitcarplay.com/blogs/autokit-blogs/product-u2w-plus
